### PR TITLE
refactor: Convert Sheet stories to csf format

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Converted stories for SheetHeader and SheetBottom 

## **Screenshots/Recordings**
Simulator Screen Shot - iPhone 14 Pro - 2023-10-10 at 18 46 02
Simulator Screen Shot - iPhone 14 Pro - 2023-10-10 at 18 46 06
Simulator Screen Shot - iPhone 14 Pro - 2023-10-10 at 18 46 09
Simulator Screen Shot - iPhone 14 Pro - 2023-10-10 at 18 46 12

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by tommasini — DISMISSED** [review]
LGTM!

**Comment by tommasini** [comment]
Curious question, what happens if the name of the title and the top right button are bigger?

File: .storybook/main.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ module.exports = {
   stories: [
     '../app/component-library/components/Cards/Card/**/*.stories.?(ts|tsx|js|jsx)',
     '../app/component-library/components/Banners/Banner/variants/BannerAlert/**/*.stories.?(ts|tsx|js|jsx)',
+    '../app/component-library/components/Sheet/**/*.stories.?(ts|tsx|js|jsx)',
```
Comment (georgewrmarshall) [inline .storybook/main.js:5]: Can we update this folder path so it's less specific and doesn't create merge conflicts every time a story is created?

```
stories: [
    '../app/components/**/*.stories.?(ts|tsx)',
    '../app/component-library/**/*.stories.?(ts|tsx)',
  ],
```

File: app/component-library/components/Sheet/SheetBottom/SheetBottom.stories.tsx [hunk 2]
```diff
@@ -20,23 +18,26 @@ const styles = StyleSheet.create({
...
+    </SheetBottomComponent>
   );
 };
 
-storiesOf('Component Library / SheetBottom', module).add('Default', () => (
-  <SheetBottomExample />
-));
+const SheetBottomMeta = {
+  title: 'Component Library / Sheet',
+  component: SheetBottomComponent,
```
Comment (georgewrmarshall) [inline app/component-library/components/Sheet/SheetBottom/SheetBottom.stories.tsx:38]: Are we missing argTypes here? `children`, `isInteractable`, `reservedMinOverlayHeight`

**Review by georgewrmarshall — DISMISSED** [review]
LGTM! Left some non-blocking comments
- pulled branch and checked stories and contorls

Not apart of the issue but should this truncate?

Screenshot 2023-10-24 at 12 58 45 PM

File: .storybook/main.js (hunk 1 above)
Comment (georgewrmarshall) [reply .storybook/main.js:5]: As @brianacnguyen pointed out. Storybook always will create conflicts with the `.storybook/storybook.requires.js` file anyway

File: app/component-library/components/Sheet/SheetBottom/SheetBottom.stories.tsx (hunk 2 above)
Comment (brianacnguyen) [reply app/component-library/components/Sheet/SheetBottom/SheetBottom.stories.tsx:38]: these were intentionally left out

**Comment by brianacnguyen** [comment]
> Curious question, what happens if the name of the title and the top right button are bigger?

the title will overflow to multiple lines and will center around start and end accessories

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**